src/magic_folder/api_cli.py

```diff
@@ -116,9 +116,14 @@ def dump_state(options):
             q.extend(s.parents_local)
         print("    {}: {}".format(snap_name, parents[:-4]), file=options.stdout)
     print("  remote snapshots:", file=options.stdout)
-    for snap_name in config.get_all_remotesnapshot_paths():
-        snap = config.get_remotesnapshot(snap_name)
-        print("    {}: {}".format(snap_name, snap), file=options.stdout)
+    for snap_name in config.get_all_snapshot_paths():
+        try:
+            cap = config.get_remotesnapshot(snap_name)
+        except KeyError:
+            cap = None
+        path_state = config.get_currentsnapshot_pathstate(snap_name)
+        print("    {}: {}".format(snap_name, cap), file=options.stdout)
+        print("    {}  {}".format(" "*len(snap_name), path_state), file=options.stdout)
```

Comment:
I suspect the primary output format should be something machine-parseable.  JSON, presumably.  Clearly that's not the prevailing convention with the existing CLI implementation so nothing actionable here for now.

Also, I thought the CLI was going to be a thin wrapper around an HTTP API.  It seems like most of the CLI is.  Maybe that's in future plans for `dump_state`?

Reply:
I'm not entirely sure what the plan for `dump-state` is, I'm just adding the additional data we are storing.

I suspect what we really want to do is change it to be by path, rather than remote and current snapshots as separate things.